driver-sync/src/test/functional/com/mongodb/client/AbstractClientEncryptionRewrapManyDataKeyProseTest.java

```diff
@@ -120,59 +122,50 @@ protected AbstractClientEncryptionRewrapManyDataKeyProseTest() {
+        getMongoClient().getDatabase("keyvault").getCollection("datakeys").drop();
+    }
+
     @AfterEach
     void cleanUp(){
-        getMongoClient().getDatabase("keyvault").getCollection("datakeys").drop();
+        clientEncryption.close();
     }
 
     @ParameterizedTest
     @MethodSource("data")
     public void rewrapWithSeparateClientEncryption(final String srcProvider, final String dstProvider) {
+        // Interactions with KMS are unreliable, so retry all operations
+        ClientEncryption retryingClientEncryption = new RetryingClientEncryption(clientEncryption);
+
         BsonDocument srcKey = MASTER_KEYS_BY_PROVIDER.get(srcProvider);
         BsonDocument dstKey = MASTER_KEYS_BY_PROVIDER.get(dstProvider);
         BsonString testString = new BsonString("test");
 
-        getMongoClient().getDatabase("keyvault").getCollection("datakeys").drop();
-
-        ClientEncryption clientEncryption1 = getClientEncryption(ClientEncryptionSettings.builder()
-                .keyVaultMongoClientSettings(getMongoClientSettingsBuilder().build())
-                .keyVaultNamespace("keyvault.datakeys")
-                .kmsProviders(KMS_PROVIDERS)
-                .build());
-
-        BsonBinary keyId = clientEncryption1.createDataKey(srcProvider, new DataKeyOptions().masterKey(srcKey));
+        BsonBinary keyId = retryingClientEncryption.createDataKey(srcProvider, new DataKeyOptions().masterKey(srcKey));
 
-        BsonBinary ciphertext = clientEncryption1.encrypt(
+        BsonBinary ciphertext = retryingClientEncryption.encrypt(
                 testString,
                 new EncryptOptions("AEAD_AES_256_CBC_HMAC_SHA_512-Deterministic").keyId(keyId));
 
-        ClientEncryption clientEncryption2 = getClientEncryption(ClientEncryptionSettings.builder()
-                .keyVaultMongoClientSettings(getMongoClientSettingsBuilder().build())
-                .keyVaultNamespace("keyvault.datakeys")
-                .kmsProviders(KMS_PROVIDERS)
-                .build());
-
-        RewrapManyDataKeyResult result = clientEncryption2.rewrapManyDataKey(
+        RewrapManyDataKeyResult result = retryingClientEncryption.rewrapManyDataKey(
                 new BsonDocument(),
                 new RewrapManyDataKeyOptions().provider(dstProvider).masterKey(dstKey));
-        assertEquals(1, result.getBulkWriteResult().getModifiedCount());
+        assertEquals(1, requireNonNull(result.getBulkWriteResult()).getModifiedCount());
 
-        assertEquals(testString, clientEncryption1.decrypt(ciphertext));
-        assertEquals(testString, clientEncryption2.decrypt(ciphertext));
+        assertEquals(testString, retryingClientEncryption.decrypt(ciphertext));
+        assertEquals(testString, retryingClientEncryption.decrypt(ciphertext));
     }
 
     @Test
     public void shouldThrowClientErrorWhenProviderIsNotSpecified() {
-        //given
-        ClientEncryption clientEncryption = getClientEncryption(ClientEncryptionSettings.builder()
```

Comment:
Now created in setup. But we don't want to wrap with RetryingClientEncryption because we are actually testing that the operation throws an exception.